extensions/parquet/table/src/main/java/io/deephaven/parquet/table/layout/ParquetMetadataFileLayout.java

```diff
@@ -67,64 +69,73 @@
  */
 public class ParquetMetadataFileLayout implements TableLocationKeyFinder<ParquetTableLocationKey> {
 
-    private final File metadataFile;
-    private final File commonMetadataFile;
+    private final URI metadataFileURI;
+    private final URI commonMetadataFileURI;
 
     private final TableDefinition definition;
     private final ParquetInstructions instructions;
     private final List<ParquetTableLocationKey> keys;
-    private final SeekableChannelsProvider channelsProvider;
 
-    public ParquetMetadataFileLayout(@NotNull final File directory) {
-        this(directory, ParquetInstructions.EMPTY);
-    }
-
-    public ParquetMetadataFileLayout(
-            @NotNull final File directory,
-            @NotNull final ParquetInstructions inputInstructions) {
-        this(new File(directory, METADATA_FILE_NAME), new File(directory, COMMON_METADATA_FILE_NAME),
-                inputInstructions);
-    }
-
-    public ParquetMetadataFileLayout(
-            @NotNull final File metadataFile,
-            @Nullable final File commonMetadataFile) {
-        this(metadataFile, commonMetadataFile, ParquetInstructions.EMPTY);
+    /**
+     * Create a new {@link ParquetMetadataFileLayout} for the given {@code source} and {@code inputInstructions}.
+     *
+     * @param source The source URI for the metadata file or directory containing the metadata file.
+     * @param inputInstructions The instructions for customizations while reading.
+     * @param channelsProvider The provider for seekable channels. If {@code null}, a new provider will be created and
+     *        used for all location keys.
+     */
+    public static ParquetMetadataFileLayout create(
+            @NotNull final URI source,
+            @NotNull final ParquetInstructions inputInstructions,
+            @Nullable SeekableChannelsProvider channelsProvider) {
+        final String path = source.getRawPath();
+        final boolean isMetadataFile = path.endsWith(METADATA_FILE_URI_SUFFIX);
+        final boolean isCommonMetadataFile = path.endsWith(COMMON_METADATA_FILE_URI_SUFFIX);
+        final boolean isDirectory = !isMetadataFile && !isCommonMetadataFile;
+        final URI directory = isDirectory ? source : source.resolve(".");
```

Comment:
Can we find some documentation for this?

Reply:
I got the idea from here (this is not Java though): https://stackoverflow.com/questions/1616353/how-can-i-get-a-directory-from-a-uri and read the code.
Also, the [documentation for resolve](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#resolve-java.net.URI-) kind of explains it:
> This is done by concatenating all but the last segment of this URI's path, if any, with the given URI's path and then normalizing the result as if by invoking the normalize method.